geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/QueryPlanner.scala

```diff
@@ -210,6 +213,34 @@ case class QueryPlanner(schema: String,
     }
   }
 
+  def adaptIteratorForMapAggregationQuery(accumuloIterator: CloseableIterator[Entry[Key, Value]],
+                                          query: Query,
+                                          returnSFT: SimpleFeatureType,
+                                          decoder: SimpleFeatureDecoder): CloseableIterator[SimpleFeature] = {
+    val aggregateKeyName = query.getHints.get(MAP_AGGREGATION_KEY).asInstanceOf[String]
+
+    val maps = accumuloIterator.map { kv =>
+      decoder
+        .decode(kv.getValue.get)
+        .getAttribute(aggregateKeyName)
+        .asInstanceOf[JMap[AnyRef, Int]]
+        .asScala
+        .toMap // TODO: Slow?
```

Comment:
since, i've gotten rid of scalaz, i think i can get rid of these .toMap calls also...